Sources/MapboxCoreNavigation/RouteController.swift

```diff
@@ -633,59 +656,17 @@ extension RouteController: Router {
     }
     
     public func userIsOnRoute(_ location: CLLocation, status: NavigationStatus?) -> Bool {
-        
-        guard let destination = routeProgress.currentLeg.destination else {
-            preconditionFailure("Route legs used for navigation must have destinations")
-        }
-        
-        // If the user has arrived, do not continue monitor reroutes, step progress, etc
-        if routeProgress.currentLegProgress.userHasArrivedAtWaypoint &&
-            (delegate?.router(self, shouldPreventReroutesWhenArrivingAt: destination) ??
-                DefaultBehavior.shouldPreventReroutesWhenArrivingAtWaypoint) {
-            return true
-        }
-        
-        // If we still wait for the first status from NavNative, there is no need to reroute
-        guard let status = status ?? sharedNavigator.mostRecentNavigationStatus else { return true }
-
-        /// NavNative doesn't support reroutes after arrival.
-        /// The code below is a port of logic from LegacyRouteController
-        /// This should be removed once NavNative adds support for reroutes after arrival. 
```

Comment:
So this functionality is gone in point release?

Reply:
Yes and no. We now rely on NN to support this logic, so technically we could support this particular feature.
Anyway, me too would prefer to do such changes in major release, but we don't have so much time to wait for it. This would also make things easier in terms of redesigning the API to fit better.

Reply:
NN won't support reroute after arrival (at least in the near future), so I've restored platform implementation